core/scripts/migrate_archives.py

```diff
@@ -0,0 +1,340 @@
+        """
+        logger.info('  - Backing up table to {}'.format(self.backup))
+        self.adapter.execute(
+            'create table {} as (select * from {})'
+            .format(self.backup, self.relation)
+        )
+        columns = self.get_renamed_columns()
+        except_str = ', '.join(o for o, _ in columns)
+        rename_str = ', '.join('{} as {}'.format(o, n) for o, n in columns)
+        sql = 'select * EXCEPT({}), {} from {!s}'.format(
+            except_str, rename_str, self.relation
+        )
+        self.adapter.connections.create_table(
+            database=self.relation.database,
+            schema=self.relation.schema,
+            table_name=self.relation.identifier,
+            sql=sql
+        )
+
+    def migrate_archive_table(self):
+        logger.info('  - Starting table migration')
+
+        if self.adapter.type() == 'bigquery':
+            self.migrate_archive_bigquery()
+        elif self.adapter.type() == 'snowflake':
+            self.migrate_archive_snowflake()
+        else:
+            self.migrate_archive_postgres()
+        logger.info('  - Finished table migration')
+
+    def build_archive_data(self):
+        source_relation = [
+            self.adapter.quote_as_configured(
+                self.archive_def['source_database'], 'database'
+            ),
+            self.adapter.quote_as_configured(
+                self.archive_def['source_schema'],
+                'schema'
+            ),
+            self.adapter.quote_as_configured(
+                self.archive_def['source_table'],
+                'identifier'
+            ),
+        ]
+
+        kwargs = {
+            'target_database': self.archive_def['target_database'],
+            'target_schema': self.archive_def['target_schema'],
+            'updated_at': self.archive_def['updated_at'],
+            'strategy': 'timestamp',
+            'unique_key': self.archive_def['unique_key'],
+        }
+
+        return ARCHIVE_TEMPLATE.format(
+            source_relation='.'.join(source_relation),
+            kwargs=repr(kwargs),
+            name=self.archive_def['target_table']
+        )
+
+    def write_file(self):
+        logger.debug('  - Writing snapshot to {}'.format(self.archive_path))
+        contents = self.build_archive_data()
+
+        wrote = system.make_file(path=self.archive_path, contents=contents)
+        if wrote:
+            logger.info('  - Wrote new snapshot file to {}'
+                        .format(self.archive_path))
+        else:
+            logger.error('  - Error: Could not write new snapshot file to {}'
+                         .format(self.archive_path))
+
+
+class MigrationManager(object):
+    def __init__(self, parsed, config):
+        self.args = parsed
+        self.config = config
+        log_path = getattr(self.config, 'log_path', None)
+        initialize_logger(parsed.debug, log_path)
+
+        self.archive_root = os.path.normpath(self.config.archive_paths[0])
```

Comment:
Is this going to work when we rename `archive` to `snapshot`? I'd definitely rather call the config `snapshot-paths` in dbt, and would be happy to hard-code `snapshots/` as the path here! What do you think?

Reply:
If this PR merges before the archives -> snapshots change (likely, given that these are the only comments!), we should change it in that PR.